src/xellgrid.widget.ts

```diff
@@ -429,7 +432,44 @@ class DataLayer extends Backbone.View{
       </div>
  `);
 
-    this.window_dropdown_menu.appendTo(this.toolbar);
+  const fileInput = $(`<input type="file" id="file_input"  data-instance-id="${this.data_layer.title}" accept=".csv" style="display:none">`);
+ 
+  $(document).off("click", `#open_file${this.data_layer.title}`).on("click", `#open_file${this.data_layer.title}`, () => {
+    // Trigger click event on the hidden file input element using HTMLElement's click() method
+    (fileInput[0] as HTMLInputElement).value = '';
+    (fileInput[0] as HTMLInputElement).click();
+  });
+
+  fileInput.off("change").on("change", (event: JQuery.ChangeEvent) => {
+    const file = (event.target as HTMLInputElement).files?.[0];
+    if (file) {
+      sendFileToBackend(file, this.data_layer.title);
+    }
+  });
```

Comment:
I think this will work if the size of the file is not too big. If it is too big, we probably need to find a way to compress (e.g. compress to PyArrow) and then send.